src/utils.js

```diff
@@ -0,0 +1,112 @@
+'use strict'
+
+const async = require('async')
+const fs = require('fs')
+const hat = require('hat')
+const os = require('os')
+const path = require('path')
+const exec = require('./exec')
+
+const join = path.join
+
+const isWindows = os.platform() === 'win32'
+
+// taken from https://github.com/hughsk/flat
+exports.flatten = (target) => {
```

Comment:
Why not use an util package from NPM, like `lodash.flatten`?

Reply:
[lodash.flatten](https://lodash.com/docs/4.17.4#flatten) only works on arrays, `ipfs config` expects a json path i.e. `ipfs config a.b.c 'val' --json`.